app/src/handlers/aggregate/alert/AlertWeaponAggregate.ts

```diff
@@ -0,0 +1,86 @@
+import AggregateHandlerInterface from '../../../interfaces/AggregateHandlerInterface';
+import DeathEvent from '../../census/events/DeathEvent';
+import {getLogger} from '../../../logger';
+import {inject, injectable} from 'inversify';
+import MongooseModelFactory from '../../../factories/MongooseModelFactory';
+import {TYPES} from '../../../constants/types';
+import ApplicationException from '../../../exceptions/ApplicationException';
+import {AlertWeaponAggregateSchemaInterface} from '../../../models/aggregate/AlertWeaponAggregateModel';
+
+@injectable()
+export default class AlertWeaponAggregate implements AggregateHandlerInterface<DeathEvent> {
+    private static readonly logger = getLogger('AlertWeaponAggregate');
+
+    private readonly factory: MongooseModelFactory<AlertWeaponAggregateSchemaInterface>;
+
+    constructor(@inject(TYPES.alertWeaponAggregateFactory) factory: MongooseModelFactory<AlertWeaponAggregateSchemaInterface>) {
+        this.factory = factory;
+    }
+
+    public async handle(event: DeathEvent): Promise<boolean> {
+        AlertWeaponAggregate.logger.debug('AlertWeaponAggregate.handle');
+
+        // Create initial record if doesn't exist
+        if (!await this.factory.model.exists({
+            alert: event.alert.alertId,
+            weapon: event.attackerWeaponId,
+        })) {
+            await this.insertInitial(event);
+        }
+
+        const documents = [];
+
+        if (!event.isTeamkill && !event.isSuicide) {
```

Comment:
Forgot to mention this in the other pr. ps2census v0.3.1 has now a computed property called `kill_type` which returns a Kill enum with the values Normal, Teamkill, Suicide, and Undetermined(freaking NSO).